### PR TITLE
Fixes #6912: Update Rudder authentication to allows plugins

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/6912

### CODE REVIEW

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala [hunk 1]
```diff
@@ -134,6 +186,27 @@ class AppConfigAuth extends Loggable {
     }
   }
 
+  @Bean def fileAuthenticationProvider : AuthenticationProvider = {
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala]: There is no ldapAuthenticationProvider ?

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala [hunk 2]
```diff
@@ -185,6 +192,23 @@ object RudderProperties {
       case FileSystemResource(file) => ConfigFactory.load(ConfigFactory.parseFile(file))
     }
   }
+
+  //some logic for the authentication providers
+  val authenticationMethods: Seq[AuthenticationMethods] = {
+    val names = try {
+      //config.getString can't be null by contract
+      config.getString("rudder.auth.type").split(",").toSeq.map( _.trim).collect { case s if(s.size > 0) => s}
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala]: Is the property fetched from the configuration.property file ? if so it misses the updated configuration file  (and we should have a migration script)

File: rudder-web/pom.xml [hunk 3]
```diff
@@ -78,7 +80,7 @@ along with this program. If not, see <http://www.gnu.org/licenses/agpl.html>.
       <groupId>com.normation.rudder</groupId>
       <artifactId>rudder-core</artifactId>
       <version>${rudder-version}</version>
-    </dependency>
+     </dependency>
```
Comment (VinceMacBuche) [inline rudder-web/pom.xml]: I Think this space should not be added (not aligned with the 'dependency' below)

File: rudder-web/src/main/scala/bootstrap/liftweb/LiftInitContextListener.scala [hunk 4]
```diff
@@ -105,32 +103,4 @@ class LiftInitContextListener extends ContextLoaderListener {
     super.contextDestroyed(sce)
   }
 
-  private[this] def initSpringAuthentication(): Unit = {
-    import RudderProperties.{config => c}
-
-    val springConfigFile = if( c.getBoolean("rudder.auth.ldap.enable")) {
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/bootstrap/liftweb/LiftInitContextListener.scala]: That property should be removed from configuration file but myabe it is already removed ;)

**Comment by VinceMacBuche** [comment]
@fanf I made some feedback can you look at them ?.

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala (hunk 2 above)
Comment (fanf) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala]: Yes, you're right

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala (hunk 1 above)
Comment (fanf) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala]: No, it is still defined in rudder-web/src/main/resources/applicationContext-security-auth-ldap.xml

Comment (fanf) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala]: So you can see what you need to either defined a code-based provider, or a file based one. And you get a tast of the horrors of spring.

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala (hunk 2 above)
Comment (fanf) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala]: Added in last PR update

**Comment by fanf** [comment]
PR updated

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala (hunk 1 above)
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala]: That seems great ( I mean great like the great depression of 1929)

**Comment by VinceMacBuche** [comment]
Should we wait for the migration script to be done before merging this ?

**Comment by fanf** [comment]
Yes, we should, else most likely it will lead to inconsistencies. But not sure when I will be able to do it.

**Comment by fanf** [comment]
Merging this one since the migration script was merged.

**Pull request merged**